### PR TITLE
Init Docker build

### PR DESCRIPTION
**Pull request by lazypower**
- Adds the dockerfile assets from vapor-ware/netbox-docker
- Slight changes to keep the root directory clean (nesting dirs in
  docker path)
- Adds a rudimentary job label to build on micro-k8s builder
- adds docker build/publish stages to the pipeline for branch builds.

### Fixes:

nothing, this only adds features that were missing. :cake:

### CODE REVIEW

File: .jenkins [hunk 1]
```diff
@@ -1,16 +1,39 @@
+def getSafeBranchName() {
```
Comment (marcoceppi) [inline .jenkins]: Not for this review, but we use these two methods I wrote ages ago in a lot of pipelines, we should try to figure out how to write these methods once and make them a plugin or a shared library.

**Comment by lazypower** [comment]
note: The ldap_configuration.py is not used in this docker build routine. I left it in-tact incase we decide to support it, but i'm not opposed to omitting it from the git-go if we know we wont be doing an ldap backend.

in either case, i think this is ready for review.

File: .jenkins [hunk 2]
```diff
@@ -1,16 +1,39 @@
...
+	return "${env.BRANCH_NAME}".replace('/', '-')
+}
+
+def getTagName() {
+	def branchName = getSafeBranchName()
+	if ("${env.BRANCH_NAME}" == 'development') {
```
Comment (marcoceppi) [inline .jenkins]: We can drop the build stuff, that was only for the KE portal. Instead we use the `develop` and `stable` branches for code promotion

File: docker/docker-entrypoint.sh [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    SUPERUSER_PASSWORD="$(< /run/secrets/superuser_password)"
+  else
+    SUPERUSER_PASSWORD='admin'
+  fi
+fi
+if [ -z ${SUPERUSER_API_TOKEN+x} ]; then
+  if [ -f "/run/secrets/superuser_api_token" ]; then
+    SUPERUSER_API_TOKEN="$(< /run/secrets/superuser_api_token)"
+  else
+    SUPERUSER_API_TOKEN='0123456789abcdef0123456789abcdef01234567'
```
Comment (marcoceppi) [inline docker/docker-entrypoint.sh]: I just realized we need to make sure we set the `SUPERUSER_API_TOKEN` in the helm chart, as this is not a very safe thing

File: .gitignore [hunk 4]
```diff
@@ -11,8 +11,9 @@
 !upgrade.sh
 fabfile.py
 *.swp
-gunicorn_config.py
 .DS_Store
 .vscode
 .venv
 .tox
+gunicorn_config.py
```
Comment (marcoceppi) [inline .gitignore]: Lets move this back to where it was, it'll cut down on upstream sync conflicts

**Pull request merged**